splice_machine/src/main/java/com/splicemachine/derby/stream/function/StreamFileFunction.java

```diff
@@ -57,9 +56,9 @@ public void readExternal(ObjectInput in) throws IOException, ClassNotFoundExcept
     }
 
     @Override
-    public Iterable<LocatedRow> call(final InputStream s) throws Exception {
+    public Iterator<LocatedRow> call(final InputStream s) throws Exception {
         if (operationContext.isFailed())
-            return Collections.emptyList();
+            return Collections.<LocatedRow>emptyList().iterator();
         checkPreference();
 
         return new Iterable<LocatedRow>() {
```

Comment:
We can remove this Iterable, just return the iterator we are creating in line 69 directly.